Added BirthCode class for additional birth codes

pascalvenema opened the pull request:
https://github.com/som-shahlab/femr/issues/178

mmcdermott reviewed (commented):
Unfortunately, I don't think we want to go in this direction; MEDS is not anchored to a particular vocabulary, so we don't necessarily want to change the birth code to reflect just the SNOMED vocabulary, and it is also important that the code be a simple string, not a python object as the primary interface for MEDS is data driven. Can you elaborate a bit in a GitHub issue what motivated this change, so we can see if there are alternative solutions?

pascalvenema commented:
Hi @mmcdermott,

Thank you for the elaboration. The reason for this change is that there is currently one SNOMED birth code 'hardcoded' into MEDS. However, there are multiple SNOMED codes that represent a birth (like the one I added via the class class construction). In my work, I cannot use the MEDS library because no birth code is found for certain patients that have the different birth code. 

I fixed it by monkeypatching the solution I proposed in this pull request. I agree with you that it is probably a good idea to not only support the SNOMED vocabulary, but this may require some added flexibility in the way the codebase is set up.

EthanSteinberg commented:
@mmcdermott  @pascalvenema  I think there is some confusion around older versions of MEDS, which had a hardcoded SNOMED birth code (https://github.com/Medical-Event-Data-Standard/meds/blob/0.2.0/src/meds/schema.py#L21) and the current version, which doesn't. 

@pascalvenema I think your concern has already been addressed with the MEDS v3 release with the introduction of MEDS_BIRTH as a generic placeholder that should work across datasets.
